File: algorithms/36. Valid Sudoku/solution.cc
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        bool r[9][9] = {0}, c[9][9] = {0}, g[3][3][9] = {0};
        for ( int i = 0; i < 9; ++i ) {
            for ( int j = 0; j < 9; ++j ) {
                if ( board[i][j] != '.' ) {
                    int n = board[i][j] - '1';
                    if ( r[i][n] || c[j][n] || g[i/3][j/3][n] ) {
                        return false;
                    }
                    r[i][n] = c[j][n] = g[i/3][j/3][n] = true;
                }
            }
        }
        return true;
    }
};

File: algorithms/36. Valid Sudoku/solution.py
class Solution(object):
    def isValidSudoku(self, board):
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        unique = []
        for i, row in enumerate(board):
            for j, val in enumerate(row):
                if val != '.':
                    unique += [(i, val), (val, j), (i/3, j/3, val)]
        return len(unique) == len(set(unique))
